backend/app/Http/Controllers/TransactionController.php

```diff
@@ -72,6 +73,11 @@ public function index(Request $request, Account $account = null)
             $transactions = $transactions->where('transaction_type', $request->type);
         }
 
+        // Filter transactions by date
+        if (strtotime($request->from) && strtotime($request->to)) {
```

Comment:
```suggestion
        if (strtotime($validated['from']) && strtotime($request->to)) {
```
i have overlooked the previous filter. lets add validation rules for the fields here

rule for date
and `to` must be required if from is present

Reply:
Added validation rule